jamesb93/mtriage
# Suggested changes

Hey there, I took a look at your PR today and figured this was the easiest way to make some suggestions and indicate how to move forward. 

## Steps to test

1. Renamed 'fourchan' to 'Fourchan', to keep with the updated format for components.
 
2. Create a new file 'data/fourchan-test.yml' that looks like this:
```yaml
folder: media/fourchan_protests
select:
  name: Fourchan
  config:
    dev: true
    board: "v"
    max_posts: 2
```

3. Create a new whitelist in preparation for a custom build 'whitelists/only-fourchan.txt', that just contains a single line 'Fourchan'.

4. Create a custom build with the whitelist using the following command:
```bash
./mtriage dev build --tag fourchan --whitelist whitelists/only-fourchan.txt
```

5. Test the analyser using the custom build and test config:
```bash
./mtriage run data/fourchan-test.yml --tag fourchan --dev
```

I had to rebase your branch on top of master, and then make a couple of very simple adjustments to the code in core.py to get it to run, see the additional commit in this PR. 

## Comments

In general this is really good work! There are a couple of things that are unclear to me.
- It doesn't seem as though the `max_posts` attribute is limiting much. As far as I can tell, the analyser goes on downloading until I cut it off.
- An 'element' in mtriage is designed to represent a single media element from some source. In fourchan's case, I think it would make sense for a single element to represent a single post. Thus, what is returned from `analyse_element` should be an element that contains _all_ the images, comments, and original text on a post. Does this make sense to you?
